src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -147,11 +147,23 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             // Set the caret position to the properly indented column in the desired line.
             var caretPosition = GetIndentedLinePosition(newDocument, newDocument.Text, desiredCaretLine.LineNumber, options, cancellationToken);
 
-            // The new line edit is calculated against the original text, d0, to get text d1.
-            // The formatting edits are calculated against d1 to get text d2.
-            // Merge the formatting and new line edits into a set of whitespace only text edits that all apply to d0.
-            var overallChanges = newLineEdit != null ? GetMergedChanges(newLineEdit.Value, formattingChanges, newDocument.Text) : formattingChanges;
-            return new BraceCompletionResult(overallChanges, caretPosition);
+            return new BraceCompletionResult(GetOverallChanges(), caretPosition);
+
+            ImmutableArray<TextChange> GetOverallChanges()
+            {
+                // The new line edit is calculated against the original text, d0, to get text d1.
+                // The formatting edits are calculated against d1 to get text d2.
+                // Merge the formatting and new line edits into a set of whitespace only text edits that all apply to d0.
+                if (!newLineEdit.HasValue)
+                    return formattingChanges;
+
+                // Depending on options, we might not get any formatting change.
+                // In this case, the newline edit is the only change.
+                if (formattingChanges.IsEmpty)
+                    return ImmutableArray.Create(newLineEdit.Value);
```

Comment:
consdier moving this into GetMergedChanges.

Reply:
Done